File: library/sync/src/main/java/org/cru/godtools/sync/job/SyncSharesJob.java
package org.cru.godtools.sync.job;

import android.support.annotation.NonNull;

import org.cru.godtools.sync.task.ToolSyncTasks;

public final class SyncSharesJob extends BaseSyncJob {
    static final String TAG = "sync_shares";

    @NonNull
    @Override
    protected Result onRunJob(final Params params) {
        final boolean result = new ToolSyncTasks(getContext()).syncShares();
        return result ? Result.SUCCESS : Result.RESCHEDULE;
    }

    public static void scheduleSharesJob() {
        scheduleSyncJob(TAG);
    }
}

File: library/sync/src/main/java/org/cru/godtools/sync/job/SyncSharesJob.kt
package org.cru.godtools.sync.job

import com.evernote.android.job.Job
import org.cru.godtools.sync.task.ToolSyncTasks

internal const val SHARES_JOB_TAG = "sync_shares"

fun scheduleSharesJob() {
    scheduleSyncJob(SHARES_JOB_TAG)
}

internal class SyncSharesJob : BaseSyncJob() {
    override fun onRunJob(params: Job.Params): Job.Result {
        val result = ToolSyncTasks(context).syncShares()
        return if (result) Job.Result.SUCCESS else Job.Result.RESCHEDULE
    }
}
